### ISSUE TITLE
Mix InputSources in Launchers (jar + java)

### ISSUE THREAD

**Issue by Guchman** [INRIA/spoon#3106]
Hi, 
maybe it will be better to have API like this:
```java
launcher.addInputResource(new JarResource(linkToJar))
launcher.addInputResource(new SourceResource(linkToJavaSource))
```
I doesn't find option to add external libraries smoothly. And make Jar/Source/etc.  as InputResource level seems to be a right abstraction, am I right?

**Comment by monperrus** [comment]
yes! this is exactly what we have:

```java
launcher.addInputResource(new ZipFile(linkToJar))
launcher.addInputResource(new FileSystemFolder(linkToJavaSource))
```

**Comment by Guchman** [comment]
So, what the  purpose of JdkLoader then?

**Comment by monperrus** [comment]
`new ZipFile(linkToJar)` analyzes the source files in the JAR.

`JdkLoader` analyzes the binary files in the JAR.

It would be great that 
`JdkLoader implements SpoonResource`. @nharrand WDYT?

**Comment by nharrand** [comment]
We don't have any class called `JdkLoader`. (So, my apologies if I missunderstood the question.)
But, if what you want to do is to add resources that contain in fact bytecode (and not sources), today, you may add them either in the classpath, or through the `JarLauncher`.
When added through the classpath, some information about the element contain in the bytecode may be accessed through reflection. When using the `JarLauncher`, a decompiler is called and then, the decompiled sources are analyzed.
If the question is: Should we have a `SpoonRessource` that call the decompiler? Yes I can add that.

**Comment by nharrand** [comment]
Implemented in #3124 ! See [DecompiledResource](https://github.com/INRIA/spoon/blob/master/spoon-decompiler/src/main/java/spoon/DecompiledResource.java).
